packages/block-library/src/gallery/block.json

```diff
@@ -74,8 +30,18 @@
 		"sizeSlug": {
 			"type": "string",
 			"default": "large"
+		},
+		"imageSettings": {
+			"type": "object",
+			"default": {
+				"allowResize": false,
+				"allowAlign": false
+			}
 		}
 	},
+	"providesContext": {
+		"gallery/imageSettings": "imageSettings"
```

Comment:
This isn't how you should use context namespaces. The namespace should be plugin/vendor-level. In this case, since it's core, there's no need for a namespace. (Existing block contexts created in core code don't use namespaces.)

Additionally, I don't think this should be a single context; It should be two separate ones. The `allowAlign` one could be reused in other blocks, rather than being tied to an `imageSettings` object.

I'd call the two contexts "allowBlockAlign" (to avoid any potential future conflict with contexts for text/content alignment) and "allowResize".

Reply:
ahh, thanks for the explanation, haven't used contexts before, that makes sense, have made that change.